src/Octoshift/GithubClient.cs

```diff
@@ -122,6 +128,8 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
                 _log.LogVerbose($"HTTP BODY: {body.ToJson()}");
             }
 
+            AddHeaders(customHeaders);
```

Comment:
At this point probably it's easier to create a `HttpRequestMessage` and use the `HttpClient#SendAsync()` method that accepts an http request instead of adding and removing the headers every single time. This way we wouldn't need the `switch` statement and it will be less code overall 😄 this is roughly what I mean:

```c#
...
_log.LogVerbose($"HTTP {httpMethod}: {url}");

using var request = new HttpRequestMessage(httpMethod, url).AddHeaders(customHeaders);

if (body != null)
{
    _log.LogVerbose($"HTTP BODY: {body.ToJson()}");
    request.Content = body.ToJson().ToStringContent();
}

using var response = await _httpClient.SendAsync(request);

// the rest of the method body stays exactly the same
```

`request.AddHeaders()` is gonna be an extension method like below, or if you don't feel like adding one you can use the `AddHeaders()` private method. 
```c#
public static HttpRequestMessage AddHeaders(this HttpRequestMessage request, Dictionary<string, string> headers)
{
    ...
}
```

Reply:
Love this approach. Switched to this one!!!